language/move-core/types/src/state.rs

```diff
@@ -0,0 +1,21 @@
+use std::cell::RefCell;
+
+#[derive(Clone, Copy, Debug)]
+pub enum VMState {
+    DESERIALIZER,
+    VERIFIER,
+    RUNTIME,
+    OTHER,
+}
+
+thread_local! {
+    static STATE: RefCell<VMState> = RefCell::new(VMState::OTHER);
+}
+
+pub fn set_state(state: VMState) -> VMState {
```

Comment:
I wish we could make this pub(crate) if we imagine folks reading get_state

Reply:
Would `pub(crate)` work here? I think it would break access because `move-core/types` is its own crate.

Reply:
I agree that public is inevitable here as this may be needed "way outside" the Move VM